File: NCAA/scripts/kalman_state.mjs
import { fileURLToPath } from "url";
import { createKalmanState } from "../../core-js/kalman_state.mjs";

const DATA_DIR = fileURLToPath(new URL("../data", import.meta.url));

const {
  KALMAN_DEFAULTS,
  loadKalmanState,
  saveKalmanState,
  initializeKalman,
  applyDailyDrift,
  updateFromGame,
  batchUpdate,
  getTeamAdj,
  kalmanSummary,
  pruneProcessedGames,
} = createKalmanState({
  dataDir: DATA_DIR,
  initialVar: 20,
  gameNoise: 196,
  maxVar: 40,
});

export {
  KALMAN_DEFAULTS,
  loadKalmanState,
  saveKalmanState,
  initializeKalman,
  applyDailyDrift,
  updateFromGame,
  batchUpdate,
  getTeamAdj,
  kalmanSummary,
  pruneProcessedGames,
};

File: pyMLB/scripts/kalman_state.py
# pyMLB/scripts/kalman_state.py -- thin wrapper around core/kalman_state.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..'))
import defaults
import core.kalman_state as _ks
_ks.STATE_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'data', 'kalman_state.json')
_ks.KALMAN_DEFAULTS = defaults.KALMAN_DEFAULTS
from core.kalman_state import (
    load_kalman_state, save_kalman_state, initialize_kalman,
    apply_daily_drift, batch_update, kalman_summary, prune_processed_games,
)
